### ISSUE TITLE
Unable to Sign Transactions from Injected Ethereum-Style Accounts

### ISSUE THREAD

**Issue by themacexpert** [polkadot-js/extension#907]
I am unable to successfully sign any transactions in Polkadot Js Apps using an injected Ethereum-style account from the extension. 

Steps to Re-Create:
- Create a new Account in the Polkadot Js extension that uses an Ethereum-style address. Moonbase Alpha or Moonriver for Example.
- To avoid using real funds, you can get Moonbase Alpha tokens in the #moonbase-faucet channel of our discord here: https://discord.gg/moonbeam. You can request funds using the format `!faucet send <your address>`
- Go to Polkadot Js Apps and update the Metadata for the Extension.
- Try to send funds from the injected account to any address including itself. You'll get the below error.
![Screen Shot 2021-12-06 at 2 51 46 PM](https://user-images.githubusercontent.com/5834262/144935583-f861b355-4fc1-4349-a3d0-7200be458244.png)

Looking at the console I see the following messages:
![Screen Shot 2021-12-06 at 2 54 46 PM](https://user-images.githubusercontent.com/5834262/144935894-af8e3c2f-0480-49e9-b31c-efd7ccdbf9d1.png)

This issue has been replicated by several individuals with the same results.

**Comment by jacogr** [comment]
Would need to look at the Moonbeam transaction signature type, e.g. `MultiSignature` or `AnySignature`, my guess is that is where it would break down. (But would then apply to all, not really "only extension accounts")

So generally the above is either -

- signing payload incorrect
- signature is in the wrong format (e.g. MultiSignature is the signature + a single leading byte)
- ... other?

There are tricky. If it only happens on extension accounts, it would mean that the extension does something weird. (Well, could also be in the apps UI signer where it uses the signature from the extension)

TL;DR Only ramblings above... this is not something that is clear in what may be the issue.

**Comment by joelamouche** [comment]
@jacogr Any news on this?

>signing payload incorrect

I added a test to the extension repo to make sure that is not the problem

>signature is in the wrong format

How do we test that? Unfortunately this is hidden in the `.sign` function

**Comment by joelamouche** [comment]
@jacogr Any news on this? This is still a problem

**Comment by joelamouche** [comment]
I havent been able to find the solution to this but it is pretty important to allow new users to join in. I would love to get on a pair programming session

**Comment by hoale58-vl** [comment]
Hello guys, 
I'm encountering the same problem with Ethereum-style chains. Those accounts I import using PolkadotJS-app UI work fine, except for this extension.
It's very weird that last week I was still doing extrinsic calls, now all submitted extrinsics are returning bad signature errors.
I installed both versions from the chrome store and the latest version from this repository.

**Comment by hoale58-vl** [comment]
I have manually updated the ExtrinsicSignature to EthereumSignature and work fine. Looks like the extensions cannot update the metadata from Ethereum chains
